src/main/java/com/indeed/jiraactions/api/ApiCaller.java

```diff
@@ -3,178 +3,108 @@
     }
 
     public JsonNode getJsonNode(final String url) throws IOException {
-        HttpsURLConnection urlConnection = null;
-        Map<String, List<String>> headers = null;
-        BufferedReader br = null;
-        String apiResults = null;
-        try {
-            urlConnection = getURLConnection(url);
-            headers = urlConnection.getRequestProperties();
-            final InputStream in = urlConnection.getInputStream();
-            br = new BufferedReader(new InputStreamReader(in));
-            apiResults = br.readLine();
-
-            final String anodeId = urlConnection.getHeaderField("X-ANODEID");
-
-            if(!Objects.equals(pinnedNode, anodeId)) {
-                if(pinnedNode != null) {
-                    log.warn("Expected X-ANODEID={} but found {}", pinnedNode, anodeId);
-                }
-                final Map<String, List<String>> responseHeaders = urlConnection.getHeaderFields();
-                final List<String> cookies = responseHeaders.get("Set-Cookie");
-                if(cookies != null) {
-                    for (final String cookie : cookies) {
-                        if (cookie.startsWith("JSESSIONID=")) {
-                            final int start = "JSESSIONID=".length();
-                            final int end = cookie.contains(";") ? cookie.indexOf(";") : cookie.length();
-                            jsessionId = cookie.substring(start, end);
-                        } else if (cookie.startsWith("upstream")) {
-                            final int start = "upstream=".length();
-                            final int end = cookie.contains(";") ? cookie.indexOf(";") : cookie.length();
-                            upstream = cookie.substring(start, end);
-                        }
-                    }
-                    if (jsessionId != null || upstream != null) {
-                        setCookies();
-                        pinnedNode = anodeId;
-                        log.info("Set JSESSION={};upstream={}. Pinning to X-ANODEID={}",
-                                jsessionId, upstream, anodeId);
-                    }
-                }
+        final Request request = new Request.Builder()
+                .header("Authorization", this.authentication)
+                .header("Cache-Control", "no-store")
+                .header("Cookie", cookies)
+                .url(url)
+                .build();
+        final Response response = client.newCall(request).execute();
+
+        final String anodeId = response.header("X-ANODEID");
+
+        if(!Objects.equals(pinnedNode, anodeId)) {
+            if (pinnedNode != null) {
+                log.warn("Expected X-ANODEID={} but found {}", pinnedNode, anodeId);
             }
-            return objectMapper.readTree(apiResults);
-        } catch (final IOException e) {
-            final StringBuilder sb = new StringBuilder();
+            cookies = String.join(";", response.headers("Set-Cookie"));
```

Comment:
We wouldn't want to replace all the cookies we have with whatever is new, because that would be throwing away old cookies.